src/DIRAC/WorkloadManagementSystem/Optimizer/InputData.py

```diff
@@ -0,0 +1,414 @@
+"""
+The InputData Optimizer queries the file catalog for specified job input data and adds the
+relevant information to the job optimizer parameters to be used during the scheduling decision.
+"""
+import time
+
+from DIRAC import S_OK, S_ERROR
+from DIRAC.Core.Utilities.Proxy import executeWithUserProxy
+from DIRAC.ConfigurationSystem.Client.Helpers.Operations import Operations
+from DIRAC.Resources.Storage.StorageElement import StorageElement
+from DIRAC.Resources.Catalog.FileCatalog import FileCatalog
+from DIRAC.DataManagementSystem.Utilities.DMSHelpers import DMSHelpers
+from DIRAC.DataManagementSystem.Client.DataManager import DataManager
+from DIRAC.WorkloadManagementSystem.Client import JobMinorStatus
+from DIRAC.WorkloadManagementSystem.Client.JobState.JobState import JobState
+from DIRAC.WorkloadManagementSystem.OptimizerAdministrator.Optimizer import Optimizer
+
+
+class InputData(Optimizer):
+    """
+    The specific Optimizer must provide the following methods:
+      - optimize() - the main method called for each job
+    """
+
+    def __init__(self):
+        """Initialize specific parameters for InputData executor."""
```

Comment:
```suggestion
        """Initialize specific parameters for InputData optimizer task."""
```
(there are probably many examples like this one)